activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -139,6 +141,33 @@ def test_connection_pool_per_pid
           rd.close
         end
 
+        def test_forked_child_doesnt_mangle_parent_connection
+          object_id = ActiveRecord::Base.connection.object_id
+          assert ActiveRecord::Base.connection.active?
+
+          rd, wr = IO.pipe
+          rd.binmode
+          wr.binmode
+
+          pid = fork {
+            rd.close
+            if ActiveRecord::Base.connection.active?
+              wr.write Marshal.dump ActiveRecord::Base.connection.object_id
+            end
+            wr.close
+
+            exit # allow finalizers to run
+          }
+
+          wr.close
+
+          Process.waitpid pid
+          assert_not_equal object_id, Marshal.load(rd.read)
+          rd.close
+
+          assert_equal 5, ActiveRecord::Base.connection.select_value("SELECT 2 + 3")
```

Comment:
Hi, I'm testing this pull request with Oracle enhanced adapter to see how to implement `discard!`  method. As of right now what I can say is Oracle database always need `from` for every select statement.

```
ActiveRecord::StatementInvalid: OCIError: ORA-00923: FROM keyword not found where expected: SELECT 2 + 3
```
Would you consider this so that all bundled + Oracle enhanced adapter can support this test?

Reply:
Sure! I'll look through existing tests for a better idea of a query to run.

I initially used `#active?`, but that just returned false when the test failed (i.e., without the corresponding change), and I wanted to see the actual "connection is broken / server has gone away" exception.